pkg/sentry/socket/netstack/netstack.go

```diff
@@ -3111,6 +3112,14 @@ func interfaceIoctl(ctx context.Context, io usermem.IO, arg int, ifr *linux.IFRe
 		// Gets the MTU of the device.
 		hostarch.ByteOrder.PutUint32(ifr.Data[:4], iface.MTU)
 
+	case linux.SIOCSIFMTU:
+		MTU := hostarch.ByteOrder.Uint32(ifr.Data[:4])
+		if MTU >= linux.MIN_MTU && MTU <= linux.MAX_MTU {
+			iface.MTU = MTU
```

Comment:
This just sets the MTU on the Interface struct which isn't actually used by the underlying network implementation.

Each of these implementations copies out the information needed.
https://cs.opensource.google/gvisor/gvisor/+/master:pkg/sentry/socket/netstack/stack.go;l=68;drc=30bca3270b9aba3394363a7fbb81291c9b0bb898;bpv=0;bpt=1

You may need to create a SetMTU() method or something like that on the inet.Stack interface if you want this to actually be reflected on the network stack.

Reply:
I see, I also just realised that adding SetMTU also requires implementing SetMTU for all of Endpoint implementations....
I think of just returning "success" if SIOCSIFMTU tries to set the MTU to the current value (i.e. no actual change), it would be sufficient for my case.

Would you consider this change acceptable?

Reply:
Is the code blocked on the ioctl call? I think most apps would just ignore errors from ioctl calls

Reply:
Unfortunately, no, OpenVPN stops when the ioctl fails.

Reply:
Yeah, I suppose we can stub this out for now, though I don't think it would be too hard to implement it properly.

You'll need to update your PR description to reflect that it's just a stub and change the 'Fixes' to 'Updates' so that #6033 isn't closed.

Reply:
Updated.
I will look into full proper implementation once I will finish other loose ends :)